From e0b6f52948255bbf5a9ae7af4167b7cbdb5f410e Mon Sep 17 00:00:00 2001
From: Wellington Wallace <wellingtonwallace@gmail.com>
Date: Sat, 21 Nov 2020 12:03:44 -0300
Subject: [PATCH] Putting the pipeline in the ready state when no app is
 playing audio. This should help to reduce Pulseaudio cpu usage

---
diff --git a/src/pipeline_base.cpp b/src/pipeline_base.cpp
index 07672fd6..bbddf03f 100644
--- a/src/pipeline_base.cpp
+++ b/src/pipeline_base.cpp
@@ -599,9 +599,9 @@ void PipelineBase::update_pipeline_state() {
           gst_element_get_state(pipeline, &s, &p, state_check_timeout);
 
           if (s == GST_STATE_PLAYING && !apps_want_to_play()) {
-            util::debug(log_tag + "No app wants to play audio. We will pause our pipeline.");
+            util::debug(log_tag + "No app wants to play audio. We will stop our pipeline.");
 
-            gst_element_set_state(pipeline, GST_STATE_PAUSED);
+            gst_element_set_state(pipeline, GST_STATE_READY);
           }
 
           return false;
-- 
2.29.2

